tests/rkt_net_test.go

```diff
@@ -86,29 +86,28 @@ func TestNetHostConnectivity(t *testing.T) {
 	child := spawnOrFail(t, cmd)
 
 	ga := testutils.NewGoroutineAssistant(t)
+	ga.Add(2)
+
 	// Child opens the server
-	ga.Add(1)
 	go func() {
 		defer ga.Done()
-		waitOrFail(t, child, true)
+		ga.WaitOrFail(child)
 	}()
 
 	// Host connects to the child
-	ga.Add(1)
 	go func() {
 		defer ga.Done()
 		expectedRegex := `serving on`
 		_, out, err := expectRegexWithOutput(child, expectedRegex)
 		if err != nil {
 			ga.Fatalf("Error: %v\nOutput: %v", err, out)
-			return
 		}
 		body, err := testutils.HttpGet(httpGetAddr)
 		if err != nil {
 			ga.Fatalf("%v\n", err)
-			return
+		} else {
```

Comment:
testing package explicitly says that t.Fatalf() should not be called inside goroutines created by test functions.